src/main/scala/firrtl/annotations/Annotation.scala

```diff
@@ -42,6 +42,9 @@ trait Annotation extends Product {
   */
 trait NoTargetAnnotation extends Annotation {
   def update(renames: RenameMap): Seq[NoTargetAnnotation] = Seq(this)
+  override def productArity: Int = 0
+  override def canEqual(that: Any): Boolean = true
+  override def productElement(n: Int): Any = 0
```

Comment:
What do you think about instead throwing a `java.lang.IndexOutOfBoundsException`? That's what both `case object Foo` or `case class Bar()` behave with `productElement`.

Reply:
Yeah, I was thinking about that yesterday. I think you are right, I'll fix it.